src/main/scala/com/cloudant/clouseau/IndexService.scala

```diff
@@ -932,4 +914,171 @@ object IndexService {
     dirCtor.newInstance(path, lockFactory).asInstanceOf[FSDirectory]
   }
 
+  /**
+   * Returns the Option[List[List[String]]] of a `drilldown` key from given map of options if it is in correct format,
+   * otherwise throws ParseException.
+   *
+   * @param options: Map[Symbol, Any] - map of options to extract `drilldown` key
+   * @return An optional value of a `drilldown` key as `Option[List[List[String]]]`
+   * @throws ParseException
+   *
+   */
+
+  private[clouseau] def getDrilldown(options: Map[Symbol, Any]): Option[List[List[String]]] = {
+    val asListOfStrings: PartialFunction[Any, List[String]] = elementWise(
+      { case string: String => string },
+      { case (container, element) => throw new ParseException(container.toString + " contains non-string element " + element.toString) }
+    )
+
+    val asListofListsOfStrings: PartialFunction[Any, List[List[String]]] = elementWise(
+      asListOfStrings,
+      { case (container, element) => throw new ParseException("invalid drilldown query " + container.toString) }
+    )
+
+    val extractor = asListofListsOfStrings.Extractor
+    getOption[Any](options, 'drilldown) match {
+      case Some(extractor(value)) =>
+        Some(value)
+      case None =>
+        None
+    }
+  }
+
+  /**
+   * Returns the Option[value] of a `ranges` key from given map of options if it is in correct format,
+   * otherwise throws ParseException.
```

Comment:
Does this have to be mentioned explicitly?  The `@throws` annotation already supplies this information.

Reply:
Fixed in aff504c6c0fcffcb599d18800d62aae05d6c9be2